Client/PaymentRequest/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,116 @@
+                    let supportedInstruments =  try JSONDecoder().decode([PaymentRequestSupportedInstrumentsHandler].self, from: supportedInstrumentsData)
+                    
+                    for item in details.displayItems {
+                        popup.addDisplayItemLabel(message: item.label + ":  " + item.amount.value + " " + item.amount.currency + "\n")
+                        log.info(item.label)
+                    }
+                    
+                    popup.addTotalLabel(message: details.total.label + ":  " + details.total.amount.value + " " + details.total.amount.currency + "\n")
+                    
+                    popup.addButton(title: Strings.paymentRequestPay) { [weak self] in
+                        guard let self = self else {
+                            return .flyDown
+                        }
+                        guard let rewards = self.tab?.rewards, let publisher = self.tab?.publisher, let amount = Double(details.total.amount.value) else {
+                            return .flyDown
+                        }
+                        
+                        rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                          // TODO: Handle started tip process
+                            self.response = """
+                                {
+                                  "requestId": "a62c29b3-f840-47cd-b895-4573d3190227",
+                                  "methodName": "bat",
+                                  "details": {
+                                    "transaction_id": "bcbbd947-346d-439f-96b4-101bbd966675",
+                                    "message": "Payment for Brave T-Shirt!"
+                                  }
+                                }
+                            """
+                            
+                            ensureMainThread {
+                               
+                                let trimmed = self.response.removingNewlines()
+                                self.tab?.webView?.evaluateJavaScript("paymentreq_postCreate('\(trimmed)')", completionHandler: { _, error in
+                                    if error != nil {
+                                        log.error(error)
+                                    }
+                            }) }
+                        }
+                        
+                        
+                        return .flyDown
+                    }
+                    
+                    popup.addButton(title: Strings.paymentRequestCancel) { () -> PopupViewDismissType in
+                        return .flyDown
+                    }
+                    
+                    log.info("Success!")
+                } catch {
+                    log.info(error)
+                }
+                popup.showWithType(showType: .flyUp)
+            }
+        }
+    }
+}
+
+extension String {
+    func removingNewlines() -> String {
```

Comment:
This is only used in one spot. Is it necessary to have it in an extension?
Maybe it's intent is clearer in an extension I guess. Up to you :)